addons/mail/models/mail_render_mixin.py

```diff
@@ -262,16 +270,25 @@ def _render_template_qweb(self, template_src, model, res_ids,
         :notice: Experimental. Use at your own risks only.
         """
         results = dict.fromkeys(res_ids, u"")
+        if not template_src:
+            return results
 
         # prepare template variables
         variables = self._render_qweb_eval_context()
         if add_context:
             variables.update(**add_context)
 
+        is_restricted = not self._unrestricted_rendering and not self.env.is_admin() and not self.env.user.has_group('mail.group_mail_template_editor')
+
         for record in self.env[model].browse(res_ids):
             variables['object'] = record
             try:
-                render_result = self.env['ir.qweb']._render(html.fragment_fromstring(template_src), variables)
+                render_result = self.env['ir.qweb']._render(html.fragment_fromstring(template_src, create_parent='div'), variables, options={'raise_on_code': is_restricted})
```

Comment:
Still think this should be cleaned. Hackish currently: haven't you any mean to detecte we have multiple nodes ? When your fragment has a single node we should avoid doing string manipulation like that.